app/js/enhancedDate.js

```diff
@@ -1,5 +1,136 @@
 /* 
  * add your module definition for enhancedDate here
  * assign the module to a global variable
- * /
+ */
 
+var enhancedDate = (function () {
+	
+	"use strict";
+
+	var myDate = null;
+	var dayOfWeek = ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"];
+	var month = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"];
+
+	function isDateSet() {
+
+		if (myDate === null) {
+
+			setDefaultDate();
+			
+		} 
+	}
+
+	function setDefaultDate() {
+
+		myDate = new Date();
+	}
+
+	return {
+		
+		/* Stores the passed-in date. 
+		 * This function can take milliseconds after epoch 
+		 * or a Date object. Do some type-checking to make 
+		 * sure you have a valid type. If no value is passed, 
+		 * default your internally stored date to now.
+		 */
+		setDate: function ( date ) {
+
+			if ( typeof date === 'number' ){
+
+				myDate = new Date(date);
+
+			} else if ( date instanceof Date ) {
+
+				myDate = date;
```

Comment:
While this works, because objects are passed by reference, if the date object is modified outside the module it will modify myDate as well.